### ISSUE TITLE
Confirmation for magit-visit-ref ?

### ISSUE THREAD

**Issue by phil-s** [magit/magit#2462]
I've just written the following for my config, because I kept on typing RET in the ref manager buffer, expecting to view the HEAD commit of the branch in question, and finding that I'd checked it out instead.

With a large repository + old refs + many tracked buffers visited, this mistake is sufficiently time-consuming to perform-and-then-revert that I think some protection (or optional protection) might be a worthwhile addition. (Personally I don't intentionally switch branches this way so often as to make this added keystroke any kind of annoyance.)

Failing that, the following code can be used:

	;; I keep unintentionally checking out branches from the ref screen.
	(defadvice magit-visit-ref (around my-protect-accidental-checkout)
	  "Protect against expensive accidental checkout of a reference."
	  (if (eq major-mode 'magit-refs-mode)
		  (if (y-or-n-p "Checkout reference? ")
			  ad-do-it
			(message "Aborted."))
		ad-do-it))
	(ad-activate 'magit-visit-ref)

**Comment by tarsius** [comment]
What about

```lisp
(define-key magit-branch-section-map "\r" 'magit-show-commit)
(define-key magit-tag-section-map "\r" 'magit-show-commit)
```

**Comment by tarsius** [comment]
Let me know if it doesn't, but I don't intend to make this function ask for confirmation. I agree with you that `RET` shouldn't checkout the branch at point, but many users disagreed. So I went back to the v1 behavior (In v2.1 it did "focus" on the branch at point, like `C-u RET` does now. That behavior makes more sense when `magit-refs-show-commit-count` is t, but I had to change the default to nil because of performance concerns.
